File: contents/data_structures/code/c/stack.c
#include<stdio.h>

int stack[256];
int top = -1;

int push(int item){
    top++;
    stack[top] = item;
}

int pop(){
    int item;
    item = stack[top];
    top--;
    return item;
}

int peek(){
    return stack[top];
}

int main(){
    push(3);
    push(5);
    push(9);
    pop();
    printf("%d\n",peek());
}

File: contents/stacks_and_queues/code/python/stack.py
#!/usr/bin/env python3

__author__ = "Michael Ciccotosto-Camp"

from typing import TypeVar, Generic


T = TypeVar("T")


class Stack(Generic[T]):
    def __init__(self) -> None:
        self.__list: list[T] = []

    def pop(self) -> T:
        return self.__list.pop()

    def push(self, element: T) -> int:
        self.__list.append(element)
        return len(self)

    def top(self) -> T:
        return self.__list[-1]

    def __len__(self) -> int:
        return len(self.__list)

    def __str__(self) -> str:
        return str(self.__list)


def main() -> None:
    int_stack: Stack[int] = Stack()

    int_stack.push(4)
    int_stack.push(5)
    int_stack.push(9)

    print(int_stack.pop())
    print(len(int_stack))
    print(int_stack.top())


if __name__ == "__main__":
    main()
